### ISSUE TITLE
Use nn.NLLLoss instead of CrossEntropyLoss

### ISSUE THREAD

**Issue by HarikrishnanBalagopal** [dev-chauhan/PQG-pytorch#7]
https://github.com/dev-chauhan/PQG-pytorch/blob/e880a4cf7341bc1e92428bff067a20bc53f639b1/score.py#L102

The `nn.CrossEntropyLoss` expects logits, not log of the probabilities.
`This criterion combines nn.LogSoftmax() and nn.NLLLoss() in one single class.`
https://pytorch.org/docs/stable/nn.html#torch.nn.CrossEntropyLoss

Use just `nn.NLLLoss` https://pytorch.org/docs/stable/nn.html#torch.nn.NLLLoss when working with log probabilties. Fix:
```
loss = nn.NLLLoss(ignore_index=data.PAD_token) 
```

**Comment by dev-chauhan** [comment]
logsoftmax(logsoftmax(x)) == logsoftmax(x)
So it should not be any different theoretically, but it can increase computation. So I will take your suggestion into account and will try using this too.
